Fix and standardize parameters.md — notation, tables, units, and row numbering

ayushman1210 opened the pull request:
##Summary:

- Cleaned and standardized [parameters.md]
- Fixed notation, LaTeX unit formatting, and inconsistent subscripts/superscripts.
- Consolidated and renumbered run-time parameter tables (rows 1–79).
- Added missing parameter symbols and parameter names; clarified definitions and notes.

##What I changed (high level):

- **Notation**
- Added/matched flux symbols (Rh, Ra, ET, T) and parameter symbol SLW.
- Added superscript exception note and X_HC subscript definition.
- **Tables and formatting**
- Converted mixed/broken sections to consistent Markdown long tables.
- Standardized use of \text{...} in LaTeX math and fixed malformed subscripts (e.g., K_{VPD,exp}, Q_{10,\text{soil}}).
- Fixed units to LaTeX math format throughout.
- **Run-time parameters**
- Sequential Row column added and numbered 1..79 for all run-time parameters.
- Assigned parameter configuration names (camelCase) where missing.
- Consolidated autotrophic/heterotrophic respiration, photosynthesis, litter, allocation, nitrogen, methane, water, and tree physiology sections.
-**Minor**
- Clarified notes on units and internal conversions (e.g., year^-1 → per-day conversion).
- Kept Hard-coded Values section unchanged except formatting consistency.


fixes #81

dlebauer commented on docs/parameters.md:
```diff
@@ -49,244 +51,197 @@ pools are mass per ground area and rates are mass per area per day. The actual p
+| 43  |                             | coarseRootExudation   | Fraction of $GPP$ from coarse roots exuded to soil[^exudates]   | unitless | Pulsing parameters |
 
-[^exudates]: Fine and coarse root exudation are calculated as a fraction of GPP, but the exudates are subtracted from the fine and coarse root pools, respectively. <!--Note that previous versions incorrectly defined fine root exudates as a fraction of NPP-->
+[^exudates]: Fine and coarse root exudation are calculated as a fraction of $GPP$, but the exudates are subtracted from the fine and coarse root pools, respectively.
```
keep note

mswilburn commented on src/sipnet/sipnet.c:
```diff
@@ -1126,11 +1126,11 @@ void calcSoilMaintRespiration(double tsoil, double water, double whc) {
-    fluxes.maintRespiration = envi.soilC * params.baseSoilResp * moistEffect *
-                              tempEffect * tillageEffect;
+    fluxes.soilMaintRespiration = envi.soilC * params.baseSoilResp *
+                                  moistEffect * tempEffect * tillageEffect;
```
Should fluxes.soilMaintRespiration be initialized to zero at some point prior to this?

ayushman1210 replied:
Added fluxes.soilMaintRespiration = 0.0 in the else block of calcMicrobeFluxes() (when microbes are not enabled)
Removed the misleading comment that suggested not initializing it

mswilburn commented on src/sipnet/sipnet.c:
```diff
@@ -1624,13 +1626,13 @@ void updatePoolsForSoil(void) {
     envi.microbeC += (microbeEff * fluxes.microbeIngestion + fluxes.soilPulse -
-                      fluxes.maintRespiration) *
+                      fluxes.microbeMaintRespiration) *
                      climate->length;
```
Does changing to fluxes.microbeMaintRespiration change the use of microbeEff variable in this calculation?

mswilburn commented on src/sipnet/sipnet.c:
```diff
@@ -1624,13 +1626,13 @@ void updatePoolsForSoil(void) {
-    fluxes.rSoil =
-        fluxes.maintRespiration + (1 - microbeEff) * fluxes.microbeIngestion;
+    fluxes.rSoil = fluxes.microbeMaintRespiration +
+                   (1 - microbeEff) * fluxes.microbeIngestion;
```
Does changing to fluxes.microbeMaintRespiration change the use of microbeEff variable in this calculation?

ayushman1210 replied:
No, changing to   fluxes.microbeMaintRespiration does not change how microbeEff is used in this calculation.
at line 1694-95 
fluxes.rSoil = fluxes.microbeMaintRespiration +  (1 - microbeEff) * fluxes.microbeIngestion;
The microbeEff is applied only to the second term: (1 - microbeEff) * fluxes.microbeIngestion

ayushman1210 commented on docs/parameters.md:
```diff
@@ -133,45 +140,44 @@ Run-time parameters can change from one run to the next, or when the model is st
 ### Photosynthesis parameters
 
-|     | Symbol                        | Parameter Name  | Definition                                                           | Units                                                                        | notes                                                                     |
+| Row | Symbol                        | Parameter Name  | Definition                                                           | Units                                                                        | Notes                                                                     |
```
done !!

mswilburn commented on docs/parameters.md:
Remove extra column before Symbol

ayushman1210 replied:
remove in the latest commit

mswilburn commented on docs/parameters.md:
```diff
@@ -133,107 +140,106 @@ Run-time parameters can change from one run to the next, or when the model is st
+| $R_{\text{a,wood},0}$ | baseVegResp         | Wood maintenance respiration rate at $0^\circ C$                         | g C respired \* g$^{-1}$ plant C \* day$^{-1}$ | read in as per-year rate only counts plant wood C; leaves handled elsewhere (both above and below-ground: assumed for now to have same resp. rate) |
+| $Q_{10v}$             | vegRespQ10          | Vegetation respiration Q10                                               |                                                | Scalar determining effect of temp on veg. resp.                                                                                                    |
+| $f_{\text{growth}}$    | growthRespFrac      | growth respiration as a fraction of recent mean NPP.                     |                                                |
+| $D_{\text{frozen}}$    | frozenSoilFolREff   | amount that foliar resp. is shutdown if soil is frozen                   |                                                | 0 = full shutdown, 1 = no shutdown                                                                                                                 |
```
Unitless?

mswilburn commented on docs/parameters.md:
```diff
@@ -133,107 +140,106 @@ Run-time parameters can change from one run to the next, or when the model is st
+| $K_{soil}$          | baseSoilResp        | Soil respiration rate at $0 ^{\circ}\text{C}$ and moisture saturated soil           | g C respired \* g$^{-1}$ soil C \* day$^{-1}$ | read in as per-year rate                              |
+| $K_{\text{meth}}$   | methOxidationRate   | Rate of methane oxidation                                                           | day$^{-1}$                                    |                                                       |
+| $Q_{10s}$           | soilRespQ10         | Soil respiration Q10                                                                |                                               | scalar determining effect of temp on soil respiration |
+| $D_{\text{moisture}}$ | soilRespMoistEffect | scalar determining effect of moisture on soil resp.                                 |                                               |                                                       |
```
Unitless?

mswilburn commented on docs/parameters.md:
```diff
@@ -133,107 +140,106 @@ Run-time parameters can change from one run to the next, or when the model is st
+| -------------------- | ------------------- | -------------------------------------------------------------------------------------------- | ------------ | -------------------------- |
+| $N_{\text{min},0}$   | mineralNInit        | Initial soil mineral nitrogen pool                                                           | $\text{g N} \cdot \text{m}^{-2}$ | Initializes $N_\text{min}$ |
+| $K_\text{vol}$       | nVolatilizationFrac | Fraction of $N_\text{min}$ volatilized per day (modulated by temperature and moisture)       | day$^{-1}$   | Eq. (17)                   |
+| $f^N_{\text{leach}}$ | nLeachingFrac       | Leaching coefficient applied to $N_\text{min}$ scaled by drainage                            | day$^{-1}$   | Eq. (18)                   |
```
Superscripts not resolving, fix formatting.

mswilburn commented on docs/parameters.md:
```diff
@@ -133,107 +140,106 @@ Run-time parameters can change from one run to the next, or when the model is st
+| $f_{\text{trans,avail}}$ | waterRemoveFrac | fraction of plant available soil water which can be removed in one day by transpiration without water stress occurring |                                                 |                                                                                                                                                |
+| $f_\text{drain,0}$       | waterDrainFrac  | fraction of plant available soil water which can be removed in one day by drainage                                     | $d^{-1}$                                        | default 1 for well drained soils                                                                                                               |
+| $D_{\text{frozen,water}}$ | frozenSoilEff   | fraction of water that is available if soil is frozen (0 = none available, 1 = all still avail.)                       |                                                 | if frozenSoilEff = 0, then shut down psn. even if WATER\_PSN = 0, if soil is frozen (if frozenSoilEff > 0, it has no effect if WATER\_PSN = 0) |
+| $WUE$                   | wueConst        | water use efficiency constant                                                                                          |                                                 |                                                                                                                                                |
```
Units?

mswilburn commented on docs/parameters.md:
```diff
@@ -133,107 +140,106 @@ Run-time parameters can change from one run to the next, or when the model is st
+| $WHC_{\text{soil}}$     | soilWHC         | soil (transpiration layer) water holding capacity                                                                      | cm                                              |                                                                                                                                                |
+| $f_{\text{intercept}}$  | immedEvapFrac   | fraction of rain that is immediately intercepted & evaporated                                                          |                                                 |                                                                                                                                                |
+| $f_{\text{fastflow}}$   | fastFlowFrac    | fraction of water entering soil that goes directly to drainage                                                         |                                                 |                                                                                                                                                |
+| $k_{\text{SOM,drain}}$  |                 |                                                                                                                        |                                                 |                                                                                                                                                |
```
Units?

mswilburn replied:
Parameter name?

mswilburn commented on docs/parameters.md:
```diff
@@ -49,61 +51,66 @@ pools are mass per ground area and rates are mass per area per day. The actual p
+| ${CH_4}_{\text{soil},0}$   |                 | Initial methane concentration in the soil                                | g C m$^{-2}$                                         |                                                                                    |
+| ${N_2O}_{\text{soil},0}$   |                 | Nitrous oxide concentration in the soil                                  | $\text{g N} \cdot \text{m}^{-2}$                   |                                                                                    |
+| $f_{\text{fine root},0}$   | fineRootFrac    | Fraction of `plantWoodInit` allocated to initial fine root carbon pool   |                                                      |                                                                                    |
+| $f_{\text{coarse root},0}$ | coarseRootFrac  | Fraction of `plantWoodInit` allocated to initial coarse root carbon pool |                                                      |                                                                                    |
```
These two are not really initial state variables. They are included as Allocation Variables, so why are they listed here? We can remove them from this table.

mswilburn commented on docs/parameters.md:
```diff
@@ -49,61 +51,66 @@ pools are mass per ground area and rates are mass per area per day. The actual p
+| $C_{\text{litter},0}$      | litterInit      | Initial litter carbon                                                    | $\text{g C} \cdot \text{m}^{-2} \text{ ground area}$ |                                                                                    |
+| $C_{\text{soil},0}$        | soilInit        | Initial soil carbon                                                      | $\text{g C} \cdot \text{m}^{-2} \text{ ground area}$ |                                                                                    |
+| $W_{\text{litter},0}$      | litterWFracInit |                                                                          | unitless                                             | fraction of litterWHC                                                              |
+| $W_{\text{soil},0}$        | soilWFracInit   |                                                                          | unitless                                             | fraction of soilWHC                                                                |
```
Add a row and details for snowInit

mswilburn commented on docs/parameters.md:
```diff
@@ -133,107 +140,106 @@ Run-time parameters can change from one run to the next, or when the model is st
+| $f_{\text{fall}}$      | fracLeafFall     | Additional fraction of leaves that fall at end of growing season        | unitless                                                |                                                |
+| $\alpha_{\text{leaf}}$  | leafAllocation   | Fraction of $NPP$ allocated to leaf growth                              | unitless                                                |                                                |
+| $K_{\text{leaf}}$      | leafTurnoverRate | Average turnover rate of leaves                                         | $\text{year}^{-1}$                                      | Converted to per-day rate internally           |
+| $LAI_{\text{max}}$     | laiMax           | Maximum leaf area index                                                 | $\text{m}^2 \text{ leaf} \cdot \text{m}^{-2} \text{ ground}$ |                                                |
```
Where does laiMax come from? I don't see it in state.h

mswilburn commented on docs/parameters.md:
```diff
@@ -133,107 +140,106 @@ Run-time parameters can change from one run to the next, or when the model is st
+| --------------------------- | --------------------- | --------------------------------------------------------------- | -------- | ------------------ |
+| $f_{\text{fine root}}$      | fineRootFrac          | Fraction of wood carbon allocated to fine roots                 | unitless |                    |
+| $f_{\text{coarse root}}$    | coarseRootFrac        | Fraction of wood carbon that is coarse roots                    | unitless |                    |
+| $\alpha_{\text{fine root}}$ | fineRootAllocation    | Fraction of $NPP$ allocated to fine roots                       | unitless |                    |
```
Add a row and info for coarseRootAllocation

mswilburn commented on docs/parameters.md:
```diff
@@ -133,107 +140,106 @@ Run-time parameters can change from one run to the next, or when the model is st
+| $K_\text{litter}$   | litterBreakdownRate | rate at which litter is converted to soil / respired at 0°C and max soil moisture   | g C broken down \* g^-1 litter C \* day^-1    | read in as per-year rate                              |
+| $f_{\text{litter}}$ | fracLitterRespired  | of the litter broken down, fraction respired (the rest is transferred to soil pool) |                                               |                                                       |
+| $K_{soil}$          | baseSoilResp        | Soil respiration rate at $0 ^{\circ}\text{C}$ and moisture saturated soil           | g C respired \* g$^{-1}$ soil C \* day$^{-1}$ | read in as per-year rate                              |
+| $K_{\text{meth}}$   | methOxidationRate   | Rate of methane oxidation                                                           | day$^{-1}$                                    |                                                       |
```
Where did methOxidationRate come from? I don't see it in state.h

mswilburn commented on docs/parameters.md:
```diff
@@ -133,107 +140,106 @@ Run-time parameters can change from one run to the next, or when the model is st
-| new | $K_N$                | nFixHalfSatMinN     | Mineral N level at which fixation suppression factor $D_{N_\text{min}}$ equals 0.5           | g N m$^{-2}$ | Eq. (19a)                  |
+| Symbol               | Parameter Name      | Definition                                                                                   | Units        | notes                      |
+| -------------------- | ------------------- | -------------------------------------------------------------------------------------------- | ------------ | -------------------------- |
+| $N_{\text{min},0}$   | mineralNInit        | Initial soil mineral nitrogen pool                                                           | $\text{g N} \cdot \text{m}^{-2}$ | Initializes $N_\text{min}$ |
```
minNInit is captured in Initial State Variables table, so remove it here.

mswilburn commented on docs/parameters.md:
```diff
@@ -133,107 +140,106 @@ Run-time parameters can change from one run to the next, or when the model is st
+| $WUE$                   | wueConst        | water use efficiency constant                                                                                          |                                                 |                                                                                                                                                |
+| $WHC_{\text{litter}}$   | litterWHC       | litter (evaporative layer) water holding capacity                                                                      | cm                                              |                                                                                                                                                |
+| $WHC_{\text{soil}}$     | soilWHC         | soil (transpiration layer) water holding capacity                                                                      | cm                                              |                                                                                                                                                |
+| $f_{\text{intercept}}$  | immedEvapFrac   | fraction of rain that is immediately intercepted & evaporated                                                          |                                                 |                                                                                                                                                |
```
Add a row and details for leafPoolDepth

mswilburn commented on docs/parameters.md:
```diff
@@ -133,107 +140,106 @@ Run-time parameters can change from one run to the next, or when the model is st
+| $D_{\text{moisture}}$ | soilRespMoistEffect | scalar determining effect of moisture on soil resp.                                 |                                               |                                                       |
+| $f_{\textrm{till}}$ | tillageEff          | Effect of tillage on decomposition that exponentially decays over time              | fraction                                      | Per‑event in `events.in`; 0 = no effect               |
 
 ### Nitrogen Cycle Parameters
```
Please add another section and table after Soil Respiration and before N Cycle Parameters called Microbial Parameters and include rows for:

- microbeInit
- baseMicrobeResp
- microbeQ10
- efficiency
- maxIngestionRate
- halfSatIngestion

Alomir commented:
@ayushman1210 Thank you for taking this on! However, please focus on getting this PR and #221 in before introducing more PRs.

ayushman1210 commented:
updated parameter.md file as per the suggestion of @mswilburn 

- Superscript formatting: Units like g C m$^{-2}$ have been updated to proper LaTeX math mode (e.g., $\text{g C} \cdot \text{m}^{-2}$) for correct rendering.
- Parameter names for initial N parameters: These appear to be placeholders for future N cycle implementation, as they are not present in the codebase. The second column remains empty.
- Extra column in stoichiometry table: Already removed.
- Day of year units: Changed to "unitless" with "day of year (1–365)" in the Notes column.
- Unit formatting and missing units: All units are now in LaTeX math mode, and missing units have been added (e.g., "unitless" for dimensionless parameters).
- Removed methOxidationRate: Not in the codebase.
- Added psnTMax: Row added for maximum photosynthesis temperature.
- Added snowInit: Row added for initial snow water equivalent.
- Removed laiMax: Not in state.h.
- Removed litterWFracInit: Not in state.h.
- Added coarseRootAllocation: Row added for coarse root allocation.
- Added fineRootQ10 and coarseRootQ10: Rows added for root respiration Q10 values.
- Removed tillageEff: Not in state.h.
- Removed mineralNInit from N Cycle: Duplication removed.
- Removed nFixFracMax and nFixHalfSatMinN: Not in state.h.
- Removed waterDrainFrac: Not in state.h.
- Removed litterWHC: Not in state.h.
- Updated W1 calculation: Changed to (water/soilWHC).
- Added leafPoolDepth: Row added.
- Removed Microbial Parameters and microbeInit: Due to PR #257.
- Standardized section headers: To title case.
- Removed fineRootFrac and coarseRootFrac from initial state: Moved to allocation parameters.
- Parameter name for k_SOM,drain: This is from alternate model ideas and not implemented; left as is.


@mswilburn is there anything else needed to update ??

ayushman1210 commented on docs/parameters.md:
```diff
@@ -49,61 +51,66 @@ pools are mass per ground area and rates are mass per area per day. The actual p
+| $N_{\text{org, litter},0}$ |                 | Initial litter organic nitrogen content                                  | $\text{g N} \cdot \text{m}^{-2}$                   |                                                                                    |
+| $N_{\text{org, soil},0}$   |                 | Initial soil organic nitrogen content                                    | $\text{g N} \cdot \text{m}^{-2}$                   |                                                                                    |
+| $N_{\text{min, soil},0}$   |                 | Initial soil mineral nitrogen content                                    | $\text{g N} \cdot \text{m}^{-2}$                   |                                                                                    |
+| ${CH_4}_{\text{soil},0}$   |                 | Initial methane concentration in the soil                                | g C m$^{-2}$                                         |                                                                                    |
```
updated in the latest commit

ayushman1210 commented on docs/parameters.md:
```diff
@@ -133,107 +140,106 @@ Run-time parameters can change from one run to the next, or when the model is st
-
+| Symbol                 | Parameter Name   | Definition                                                              | Units                                                   | Notes                                          |
+| ---------------------- | ---------------- | ----------------------------------------------------------------------- | ------------------------------------------------------- | ---------------------------------------------- |
+| $D_{\text{on}}$        | leafOnDay        | Day of year when leaves appear                                          | day of year (1–365)                                    |                                                |
```
fixed it in the latest commit

dlebauer commented on docs/parameters.md:
```diff
@@ -33,7 +32,10 @@ pools are mass per ground area and rates are mass per area per day. The actual p
+|                           | $Rh$       | Heterotrophic respiration (soil/litter decomposition) |
+|                           | $Ra$       | Autotrophic respiration (vegetation respiration)       |
 |                           | $ET$       | Evapotranspiration                                     |
+|                           | $T$        | Transpiration                                          |
```
This is a good catch, but I don't want to have to label all the "T's" is T actually used for "Transpiration" anywhere, or can we delete this line?

ayushman1210 replied:
updated this also !!

dlebauer commented on docs/parameters.md:
```diff
@@ -49,60 +51,67 @@ pools are mass per ground area and rates are mass per area per day. The actual p
+| $f_{\text{fine root},0}$   | fineRootFrac    | Fraction of `plantWoodInit` allocated to initial fine root carbon pool   | unitless                                             |                                                                                    |
+| $f_{\text{coarse root},0}$ | coarseRootFrac  | Fraction of `plantWoodInit` allocated to initial coarse root carbon pool | unitless                                             |                                                                                    |
+| $\mathfrak{CH_4}_{\text{soil},0}$   |                 | Initial methane concentration in the soil                                | $\text{g C} \cdot \text{m}^{-2}$                     |                                                                                    |
+| $\mathfrak{N_2O}_{\text{soil},0}$   |                 | Initial nitrous oxide concentration in the soil                          | $\text{g N} \cdot \text{m}^{-2}$                     |                                                                                    |
```
we use fraktur for components that have not been implemented.

dlebauer commented on docs/parameters.md:
```diff
@@ -49,60 +51,67 @@ pools are mass per ground area and rates are mass per area per day. The actual p
+| $\mathfrak{CH_4}_{\text{soil},0}$   |                 | Initial methane concentration in the soil                                | $\text{g C} \cdot \text{m}^{-2}$                     |                                                                                    |
+| $\mathfrak{N_2O}_{\text{soil},0}$   |                 | Initial nitrous oxide concentration in the soil                          | $\text{g N} \cdot \text{m}^{-2}$                     |                                                                                    |
+| $W_{\text{snow},0}$        | snowInit        | Initial snow water equivalent                                            | cm water equivalent                                  |                                                                                    |
 
```
the fact that they are planned is already conveyed by the use of mathfrak font

dlebauer commented on docs/parameters.md:
```diff
@@ -123,119 +132,115 @@ Run-time parameters can change from one run to the next, or when the model is st
+| $D_{\text{frozen,water}}$ | frozenSoilEff   | fraction of water that is available if soil is frozen (0 = none available, 1 = all still avail.)                       | unitless                                                 | if frozenSoilEff = 0, then shut down psn. even if WATER\_PSN = 0, if soil is frozen (if frozenSoilEff > 0, it has no effect if WATER\_PSN = 0) |
+| $WUE$                   | wueConst        | water use efficiency constant                                                                                          | unitless                                                 |                                                                                                                                                |
+| $WHC_{\text{soil}}$     | soilWHC         | soil (transpiration layer) water holding capacity                                                                      | cm                                              |                                                                                                                                                |
+|                         | leafPoolDepth   | leaf (evaporative) pool rim thickness                                                                                  | mm                                              |                                                                                                                                                |
```
Add symbol and fix units

```suggestion
| $WHC_{\text{leaf}}$   | leafPoolDepth   | leaf/canopy evaporative pool capacity per unit leaf area (used to limit interception evaporation when enabled) | cm    | Only used when `LEAF_WATER = 1` (`--leaf-water`). defines LAI-scaled cap on interception evaporation: $immedEvap \le LAI \cdot WHC_{\text{leaf}}$. Excess throughfall goes to the soil. |
```

dlebauer commented:
A few more updates; documented the leaf-water cap and fixed units (mm-->cm) in docs and source

dlebauer commented on docs/model-structure.md:
```diff
@@ -737,10 +737,9 @@ $F^W_{\text{precip,soil}}$ is added to soil water in \eqref{eq:Braswell_A4}.
 
-Evapotranspiration ($ET$) is calculated as the sum of evaporation ($E$) and transpiration ($T$), which are defined
-below:
+Evapotranspiration ($ET$) is calculated as the sum of evaporation ($E$) and transpiration ($F^W_{\text{trans}}$). These fluxes are defined below:
```
```suggestion
Evapotranspiration ($ET$) is calculated as the sum of evaporation ($E$) and transpiration ($T=F^W_{\text{trans}}$). These fluxes are defined below:
```
